ipykernel/kernelbase.py

```diff
@@ -953,11 +955,15 @@ async def debug_request(self, stream, ident, parent):
         reply_msg = self.session.send(stream, "debug_reply", reply_content, parent, ident)
         self.log.debug("%s", reply_msg)
 
-    # Taken from https://github.com/jupyter-server/jupyter-resource-usage/blob/e6ec53fa69fdb6de8e878974bcff006310658408/jupyter_resource_usage/metrics.py#L16
     def get_process_metric_value(self, process, name, attribute=None):
         try:
             # psutil.Process methods will either return...
-            metric_value = getattr(process, name)()
+            pid = process.pid
+            p = self.processes.get(pid, None)
+            if not p:
+                self.processes[pid] = process
+                p = self.processes.get(pid)
```

Comment:
These lines are not need as we ensure the a single, reused process below

Reply:
Fixed. Thx.